### PR TITLE
[GEOT-7570] Small features that cross the dateline may be interpreted as preflipped

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOT-7570](https://osgeo-org.atlassian.net/browse/GEOT-7570) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

removed longestline filter

cleanup

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingCoordinateFilter.java [hunk 1]
```diff
@@ -47,19 +47,29 @@ class WrappingCoordinateFilter implements GeometryComponentFilter {
...
+
     /**
      * Builds a new wrapper
      *
      * @param wrapLimit Subsequent coordinates whose X differ from more than {@code wrapLimit} are
      *     supposed to be wrapping the dateline and need to be offsetted
      * @param offset The offset to be applied to coordinates to unwrap them
+     * @param mt The math transform to use to detect the wrapping
+     * @param wrapOnY If true, the wrapping is supposed to happen on the Y axis, otherwise on the X
+     * @param isPreFlipped If true, the coordinates are already flipped, so the wrapping is supposed
```
Comment (aaime) [inline modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingCoordinateFilter.java:60]:
```suggestion
     * @param isPreFlipped If true, the coordinates are already flipped, so the wrapping is supposed
```

Supposed? :rofl:

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java [hunk 2]
```diff
@@ -49,6 +49,12 @@ public class WrappingProjectionHandler extends ProjectionHandler {
 
     private static final Object LARGE_EARTH_OBJECT = new Object();
 
+    /** The user data key to indicate that the geometry was pre-flipped */
+    protected static final String PREFLIPPED_OBJECT = "PRE-FLIPPED";
+
+    /** The tolerance to consider a geometry without touching both datelines */
+    public static final double DATELINE_PROXIMITY_TOLERANCE = 1.99572;
```
Comment (aaime) [inline modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java:56]: Comment that it was obtained by trial and error.

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java [hunk 3]
```diff
@@ -240,6 +258,24 @@ public Geometry postProcess(MathTransform mt, Geometry geometry) {
         }
     }
 
+    /**
+     * In some cases the geometry is preflipped due to coordinate order and proximity to the
+     * dateline
```
Comment (aaime) [inline modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java:263]: Write an example, 3 points polygon or 2 points line that would show what pre-flipped means.

**Review by aaime — CHANGES_REQUESTED** [review]
Generally speaking looking good, see comments below about being a bit more explicit docs wise.

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingCoordinateFilter.java (hunk 1 above)
Comment (turingtestfail) [reply modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingCoordinateFilter.java:60]: fixed

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java (hunk 2 above)
Comment (turingtestfail) [reply modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java:56]: added

File: modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java (hunk 3 above)
Comment (turingtestfail) [reply modules/library/main/src/main/java/org/geotools/renderer/crs/WrappingProjectionHandler.java:263]: added

**Pull request merged**